dataflow/gemma/custom_model_gemma.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import Optional
+from typing import Sequence
+
+import apache_beam as beam
+from apache_beam.ml.inference.base import ModelHandler
+from apache_beam.ml.inference.base import PredictionResult
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.options.pipeline_options import PipelineOptions
+
+import keras_nlp
+from keras_nlp.src.models.gemma.gemma_causal_lm import GemmaCausalLM
+
+
+class GemmaModelHandler(ModelHandler[str,
+                                     PredictionResult, GemmaCausalLM
+                                     ]):
+    def __init__(
+        self,
+        model_name: str = "gemma_2B",
+    ):
+        """ Implementation of the ModelHandler interface for spaCy using text as input.
+
+        Example Usage::
+
+          pcoll | RunInference(SpacyModelHandler())
+
+        Args:
+          model_name: The Gemma model name. Default is gemma_2B.
+        """
+        self._model_name = model_name
+        self._env_vars = {}
+
+    def share_model_across_processes(self) -> bool:
```

Comment:
Could we add a docstring explaining what this does and why its always True here?

Reply:
Done